FAQ.md

```diff
@@ -69,6 +69,147 @@ However, if you do encounter this error, please try to capture a Wireshark or `t
+
+The fictional school's student database application might have contained a query that looked like this:
+```php
+$student_name = "Robert');DROP TABLE Students;--"
+
+$query = sprintf("INSERT INTO Students (name) VALUES ('%s')", $student_name);
+$result = $mysqli->query($query);
+```
+
+When formatted into the middle of this query, the maliciously crafted input string closes the quotes and finishes the statement (`Robert');`),
+then starts another one with the nefarious payload (`DROP TABLE Students;`), and causes the rest of the original query to be ignored by starting a SQL comment (`--`).
+Thus, the database server sees, and executes, three separate statements like so:
+
+```SQL
+INSERT INTO Students(firstname) VALUES ('Robert');
+DROP TABLE Students;
+--');
+```
+
+And thus the school has lost this year's student records (at least they had last years' backed up?).
+
+The original mitigation for this attack was to make sure that any untrustworthy user input was properly escaped (or "sanitized"),
+and many frameworks provided utility functions for this, such as PHP's [`mysqli::real_escape_string()`](https://www.php.net/manual/en/mysqli.real-escape-string.php) (not to be confused with the obsolete [`mysql_real_escape_string()`](https://www.php.net/manual/en/function.mysql-real-escape-string) or [`mysql_escape_string()`](https://www.php.net/manual/en/function.mysql-escape-string.php)).
+
+These would prefix any syntatically significant characters (in this case, quotation marks) with a backslash so it's less likely to affect the database server's interpretation of the query:
+
+```php
+$student_name = $mysqli->real_escape_string("Robert');DROP TABLE Students;--");
+
+/*
+    Everything is okay now as the dastardly single-quote has been inactivated by the backslash:
+    "INSERT INTO Students (name) VALUES ('Robert\');DROP TABLE Students;--');"
+*/
+$query = sprintf("INSERT INTO Students (name) VALUES ('%s')", $student_name);
+```
+
+The database server sees the backslash and knows that the single-quote is part of the string content, not its terminating character.
+
+However, this was something that you still had to _remember_ to do, making it only half a solution. Additionally, properly escaping the string requires knowledge of the current character set of the connection which is why the `mysqli` object is a required parameter 
+(or the receiver in object-oriented style). And you could always just forget to wrap the string parameter in quotes (`'%s'`) in the first place, which these wouldn't help with.
+
+Even when everything is working correctly, formatting dynamic data into a query still requires the database server to re-parse and generate a new query plan with every new variant--caching helps, but is not a silver bullet.
+
+In comes **prepared statements** to the rescue!
```

Comment:
This feels like two idioms are mixed together, though it's understandable very well. Just a tiny suggestion:

```suggestion
In come **prepared statements** to the rescue!
```